Improve socket error handling

This should be silently handled, without writing exception to log:

```
[pool-1-thread-47] ERROR svnserver.server.SvnServer - Command execution error
org.tmatesoft.svn.core.SVNException: svn: E204900: Unknown error
at svnserver.server.command.DeltaCmd$ReportPipeline$1.textDeltaChunk(DeltaCmd.java:379)
at org.tmatesoft.svn.core.io.diff.SVNDeltaGenerator.sendDelta(SVNDeltaGenerator.java:273)
at org.tmatesoft.svn.core.io.diff.SVNDeltaGenerator.sendDelta(SVNDeltaGenerator.java:187)
at svnserver.server.command.DeltaCmd$ReportPipeline.updateFile(DeltaCmd.java:357)
at svnserver.server.command.DeltaCmd$ReportPipeline.updateEntry(DeltaCmd.java:460)
at svnserver.server.command.DeltaCmd$ReportPipeline.updateDir(DeltaCmd.java:301)
at svnserver.server.command.DeltaCmd$ReportPipeline.updateEntry(DeltaCmd.java:458)
at svnserver.server.command.DeltaCmd$ReportPipeline.sendResponse(DeltaCmd.java:215)
at svnserver.server.command.DeltaCmd$ReportPipeline.complete(DeltaCmd.java:183)
at svnserver.server.command.DeltaCmd$ReportPipeline$$Lambda$30/652706217.process(Unknown Source)
at svnserver.server.step.CheckPermissionStep.process(CheckPermissionStep.java:39)
at svnserver.server.SvnServer.serveClient(SvnServer.java:150)
at svnserver.server.SvnServer.lambda$run$0(SvnServer.java:120)
at svnserver.server.SvnServer$$Lambda$8/319518541.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.SocketException: Connection reset
at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113)
at java.net.SocketOutputStream.write(SocketOutputStream.java:153)
at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
at java.io.BufferedOutputStream.write(BufferedOutputStream.java:121)
at java.io.BufferedOutputStream.write(BufferedOutputStream.java:122)
at svnserver.parser.token.StringToken.write(StringToken.java:49)
at svnserver.parser.SvnServerWriter.binary(SvnServerWriter.java:68)
at svnserver.parser.SvnServerWriter.binary(SvnServerWriter.java:63)
at svnserver.server.command.DeltaCmd$ReportPipeline$1.textDeltaChunk(DeltaCmd.java:374)
... 16 more
```

The throw SVNException position: 

```java
public OutputStream textDeltaChunk(String path, SVNDiffWindow diffWindow) throws SVNException {
    try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
        diffWindow.writeTo(stream, header, compress);
        header = false;
        writer
            .listBegin()
            .word("textdelta-chunk")
            .listBegin()
            .string(tokenId)
            .binary(stream.toByteArray())
            .listEnd()
            .listEnd();
        return null;
    } catch (IOException e) {
        throw new SVNException(SVNErrorMessage.UNKNOWN_ERROR_MESSAGE, e); 
    }   
}
```

1. Commit big single file.
2. Client svn checkout and interrupt the connection
```
: E125002: Bad Subversion URL:svn://localhost/example
git@likui:/tmp/svn$ svn co svn://localhost/example
A    example/README.md
^Csvn: E200015: Caught signal
```
Then , server throw SVNException

```
org.tmatesoft.svn.core.SVNException: svn: E204900: Unknown error
	at svnserver.server.command.DeltaCmd$ReportPipeline$1.textDeltaChunk(DeltaCmd.java:477)
	at org.tmatesoft.svn.core.io.diff.SVNDeltaGenerator.sendDelta(SVNDeltaGenerator.java:273)
	at org.tmatesoft.svn.core.io.diff.SVNDeltaGenerator.sendDelta(SVNDeltaGenerator.java:187)
	at svnserver.server.command.DeltaCmd$ReportPipeline.updateFile(DeltaCmd.java:455)
	at svnserver.server.command.DeltaCmd$ReportPipeline.updateEntry(DeltaCmd.java:554)
	at svnserver.server.command.DeltaCmd$ReportPipeline.updateDirEntries(DeltaCmd.java:395)
	at svnserver.server.command.DeltaCmd$ReportPipeline.updateDir(DeltaCmd.java:354)
	at svnserver.server.command.DeltaCmd$ReportPipeline.updateEntry(DeltaCmd.java:552)
	at svnserver.server.command.DeltaCmd$ReportPipeline.sendDelta(DeltaCmd.java:278)
	at svnserver.server.command.DeltaCmd$ReportPipeline.sendResponse(DeltaCmd.java:288)
	at svnserver.server.command.DeltaCmd$ReportPipeline.complete(DeltaCmd.java:240)
	at svnserver.server.step.CheckPermissionStep.process(CheckPermissionStep.java:43)
	at svnserver.server.SvnServer.serveClient(SvnServer.java:293)
	at svnserver.server.SvnServer.lambda$run$0(SvnServer.java:156)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.SocketException: Connection reset
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113)
	at java.net.SocketOutputStream.write(SocketOutputStream.java:153)
	at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
	at java.io.BufferedOutputStream.write(BufferedOutputStream.java:121)
	at java.io.BufferedOutputStream.write(BufferedOutputStream.java:122)
	at svnserver.parser.token.StringToken.write(StringToken.java:56)
	at svnserver.parser.SvnServerWriter.binary(SvnServerWriter.java:75)
	at svnserver.parser.SvnServerWriter.binary(SvnServerWriter.java:70)
	at svnserver.server.command.DeltaCmd$ReportPipeline$1.textDeltaChunk(DeltaCmd.java:472)
	... 16 more
```

--- a/src/main/java/svnserver/server/SvnServer.java
+++ b/src/main/java/svnserver/server/SvnServer.java
@@ -58,14 +58,16 @@ public class SvnServer extends Thread {
   private static final Logger log = LoggerFactory.getLogger(SvnServer.class);
   private static final long FORCE_SHUTDOWN = TimeUnit.SECONDS.toMillis(5);
   @NotNull
-  private static final Set<SVNErrorCode> WARNING_CODES = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(new SVNErrorCode[]{
+  private static final Set<SVNErrorCode> WARNING_CODES = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
       SVNErrorCode.CANCELLED,
       SVNErrorCode.ENTRY_NOT_FOUND,
       SVNErrorCode.FS_NOT_FOUND,
       SVNErrorCode.RA_NOT_AUTHORIZED,
       SVNErrorCode.REPOS_HOOK_FAILURE,
       SVNErrorCode.WC_NOT_UP_TO_DATE,
-  })));
+      SVNErrorCode.IO_WRITE_ERROR,
+      SVNErrorCode.IO_PIPE_READ_ERROR
+  )));
 
   @NotNull
   private final Map<String, BaseCmd<?>> commands = new HashMap<>();

--- a/src/main/java/svnserver/server/command/DeltaCmd.java
+++ b/src/main/java/svnserver/server/command/DeltaCmd.java
@@ -563,7 +563,7 @@ public final class DeltaCmd extends BaseCmd<DeltaParams> {
                         .listEnd();
                     return null;
                   } catch (IOException e) {
-                    throw new SVNException(SVNErrorMessage.UNKNOWN_ERROR_MESSAGE, e);
+                    throw new SVNException(SVNErrorMessage.create(SVNErrorCode.IO_WRITE_ERROR), e);
                   }
                 }
 

